tools/scylla-sstable-crawler.cc

```diff
@@ -0,0 +1,693 @@
+command line arguments. Sstables will be processed by the selected
+operation one-by-one (can be changed with --merge). Any number of
+sstables can be passed but mind the open file limits. Pass the full path
+to the data component of the sstables (*-Data.db). For now it is
+required that the sstable is found at a valid data path:
+
+    /path/to/datadir/{{keyspace_name}}/{{table_name}}-{{table_id}}/
+
+The schema to read the sstables is read from a schema.cql file. This
+should contain the keyspace and table definitions, as well as any UDTs
+used.
+Filtering the sstable()s to process only certain partition(s) is
+supported via the --partition and --partitions-file command line flags.
+Partition keys are expected to be in the hexdump format used by scylla
+(hex representation of the raw buffer).
+Operations write their output to stdout, or file(s). The tool logs to
+stderr, with a logger called {}.
+
+The supported operations are:
+{}
+
+Examples:
+
+# dump the content of the sstable
+$ scylla-sstable-crawler dump /path/to/md-123456-big-Data.db
+
+# dump the content of the two sstable(s) as a unified stream
+$ scylla-sstable-crawler dump --merge /path/to/md-123456-big-Data.db /path/to/md-123457-big-Data.db
+
+# generate a joint histogram for the specified partition
+$ scylla-sstable-crawler writetime-histogram --partition={{myhexpartitionkey}} /path/to/md-123456-big-Data.db
+
+# validate the specified sstables
+$ scylla-sstable-crawler validate /path/to/md-123456-big-Data.db /path/to/md-123457-big-Data.db
+)";
+    app_cfg.description = format(description_template, app_name, boost::algorithm::join(operations | boost::adaptors::transformed(
+                    [] (const auto& op) { return format("* {}: {}", op->name(), op->description()); } ), "\n"));
+
+    const auto op_list = boost::algorithm::join(operations | boost::adaptors::transformed(
+                    [] (const auto& op) { return sstring(op->name()); } ), ", ");
+
+    const auto op_help = fmt::format("operation to run, one of ({})", op_list);
+
+    app_template app(std::move(app_cfg));
+
+    app.add_options()
+        ("schema-file", bpo::value<sstring>()->default_value("schema.cql"), "file containing the schema description")
+        ("partition", bpo::value<std::vector<sstring>>(), "partition(s) to filter for, partitions are expected to be in the hex format")
+        ("partitions-file", bpo::value<sstring>(), "file containing partition(s) to filter for, partitions are expected to be in the hex format")
+        ("merge", "merge all sstables into a single mutation fragment stream (use a combining reader over all sstable readers)")
+        ("no-skips", "don't use skips to skip to next partition when the partition filter rejects one, this is slower but works with corrupt index")
+        ;
+
+    app.add_positional_options({
+        {"operation", bpo::value<sstring>()->default_value(sstring(operations.front()->name())), op_help.c_str(), 1},
```

Comment:
How does that work if there's only one positional parameter present, do we skip options with default value?

Reply:
I just checked and sadly no. I tries to use one of the sstable arguments as the operation. Not too smart :)

Reply:
So let go with `--{operation}` options as you suggested.
It will also do a better job at verifying the options' validity and will print the help message for an unsupported operation, like `--dumb`.